enterprisemediawiki/meza
# Import wikis directly from a remote server

This pull request adds a script `import-remote-wikis.sh`, which allows for importing wikis directly from remote application and database servers.

The script can either be used without any setup by simply doing `sudo bash import-remote-wikis.sh`, or you can create a `remote-wiki-config.sh` file in `/opt/meza` which eliminates most of the prompts the user must type in.

How to test this PR on an existing meza install:

1. `sudo yum install -y cifs-utils`
2. `cd /opt/meza`
3. `sudo git fetch origin`
4. `sudo git checkout import`
5. Create `remote-wiki-config.sh` by doing one of the following:
  1. `sudo cp ./scripts/config/remote-wiki-config.example.sh ./remote-wiki-config.sh` and editing the file
  2. `sudo vi remote-wiki-config.sh` and pasting in your pre-built config
7. `cd scripts`
8. `sudo bash import-remote-wikis.sh`. You should only need to enter your username and password for the remote server if you filled `remote-wiki-config.sh`

How to test this PR on a new meza install:

1. `curl -LO https://raw.githubusercontent.com/enterprisemediawiki/meza/import/scripts/install.sh`
2. `sudo bash install.sh`
3. Make sure to select `import` as the branch to install
4. `cd /opt/meza`
5. Create `remote-wiki-config.sh` by doing one of the following:
  1. `sudo cp ./scripts/config/remote-wiki-config.example.sh ./remote-wiki-config.sh` and editing the file
  2. `sudo vi remote-wiki-config.sh` and pasting in your pre-built config
7. `cd scripts`
8. `sudo bash import-remote-wikis.sh`. You should only need to enter your username and password for the remote server if you filled `remote-wiki-config.sh`

List of changes:

* Add `cifs-utils` to `yums.sh` to allow mounting share drive
* Add `import-remote-wikis.sh` to mount a share with a remote server, get the necessary files, and retrieve database dumps from a database server
* Add `getDatabaseNameFromSetup.php` to make an easy PHP script for getting the custom database name of some incoming wikis
* Add example remote wiki settings in `remote-wiki-config.example.sh`
* Add line to `.gitignore` so remote wiki settings are ignored

Issues:

Closes #243
